k8sutil/watch.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import k8sclient
+import json
+
+import pydoc
+
+PYDOC_RETURN_LABEL = ":return:"
+
+# Removing this suffix from return type name should give us event's object type. e.g., if list_namespaces()
+# returns "NamespaceList" type, then list_namespaces(watch=true) returns a stream of events with objects
+# of type "Namespace". In case this assumption is not true, user should provide return_type to Watch class's __init__.
+TYPE_LIST_SUFFIX = "List"
+
+
+class SimpleNamespace:
```

Comment:
you can probably replace this with a named tuple (see `collections.namedtuple`), if you know what fields you need (and it looks like there's only one use below, so you probably can).